### ISSUE TITLE
Global error not set when uv_fs_sendfile() fails with EAGAIN

### ISSUE THREAD

**Issue by ghost** [joyent/libuv#611]
The following sendfile() loop, when run on a non-blocking uv_tcp_t global_tcp, fails with the message "Sendfile error: no more processes".

``` c
static void send_sendfile_block() {
    uv_fs_t freq;

    int64_t left = global_transferlen;
    while(left > 0) {
        printf("Calling with left = %ld\n", left);
        uv_fs_sendfile(uv_default_loop(), &freq, global_tcp.fd, global_file, global_transferlen-left, BUFFER_SIZE > left ? left : BUFFER_SIZE, NULL);
        if(freq.result < 0) {
            printf("Sendfile error: %s", uv_strerror(uv_last_error(uv_default_loop())));
            exit(1);
        }
        left -= freq.result;
        uv_fs_req_cleanup(&freq);
    }
}
```

The actual error, according to strace, is EAGAIN:

```
  sendfile(13, 11, [4656528], 524288)     = -1 EAGAIN (Resource temporarily unavailable)
```

freq.errorno is correctly set to UV_EAGAIN, but this error is not returned by uv_last_error(uv_default_loop()).

This is on Linux and libuv commit 97c527a.

**Comment by ghost** [comment]
Ah, I found my source of confusion. The error _is_ propagated correctly, as in, the following is true:

``` c
uv_last_error(uv_default_loop()).code == UV_EAGAIN
```

My confusion came from the error string associated with EAGAIN, which is defined as "no more processes" in uv.h:79. Why was that error string chosen, instead of the more common "Operation would block" or similar?

**Comment by bnoordhuis** [comment]
Right, good catch. Fixed the error message in 0d5d650.
